### PR TITLE
integrate Codecov into CI

### PR DESCRIPTION
**Pull request by gy95**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes part of #4327 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by vincentgoat** [comment]
Removing this operation? https://github.com/kubeedge/kubeedge/blob/master/hack/lib/golang.sh#L450

Configure thresholds of coverage.

Will it be better not to show coverage commitment?

**Comment by gy95** [comment]
PTAL @vincentgoat

File: .github/codecov.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+coverage:
+  status:
+    project:
+      default:
+        # https://docs.codecov.com/docs/commit-status#target
+        target: 80%
```
Comment (vincentgoat) [inline .github/codecov.yml:6]:
```suggestion
    project:
      default:
        # https://docs.codecov.com/docs/commit-status#target
        target: 80%
        informational: true
    patch:
      default:
        target: 80%
```

Comment (gy95) [reply .github/codecov.yml:6]: fixed

**Review by vincentgoat — COMMENTED** [review]
/lgtm

**Comment by kevin-wangzefeng** [comment]
close reopen to trigger codecov
/close

**Review by vincentgoat — COMMENTED** [review]
lgtm

**Comment by gy95** [comment]
cc @fisherxu PTAL :)

**Review by fisherxu — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**